entity-registry/src/main/java/com/linkedin/metadata/aspect/validation/UserDeleteValidator.java

```diff
@@ -0,0 +1,76 @@
+package com.linkedin.metadata.aspect.validation;
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.entity.Aspect;
+import com.linkedin.identity.CorpUserInfo;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.aspect.RetrieverContext;
+import com.linkedin.metadata.aspect.batch.BatchItem;
+import com.linkedin.metadata.aspect.batch.ChangeMCP;
+import com.linkedin.metadata.aspect.plugins.config.AspectPluginConfig;
+import com.linkedin.metadata.aspect.plugins.validation.AspectPayloadValidator;
+import com.linkedin.metadata.aspect.plugins.validation.AspectValidationException;
+import com.linkedin.metadata.aspect.plugins.validation.ValidationExceptionCollection;
+import java.util.Collection;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Validator to ensure that system users (i. e: Those can come natively with DataHub) are not
+ * deleted. *
+ */
+@Slf4j
+@Setter
+@Getter
+@Accessors(chain = true)
+public class UserDeleteValidator extends AspectPayloadValidator {
+  @Nonnull private AspectPluginConfig config;
+
+  @Override
+  protected Stream<AspectValidationException> validateProposedAspects(
+      @Nonnull Collection<? extends BatchItem> mcpItems,
+      @Nonnull RetrieverContext retrieverContext) {
+
+    ValidationExceptionCollection exceptions = ValidationExceptionCollection.newCollection();
+
+    mcpItems.forEach(
+        i -> {
+          final Urn entityUrn = i.getUrn();
+
+          if (!entityUrn.getEntityType().equals(Constants.CORP_USER_ENTITY_NAME)) {
```

Comment:
Entity type filter not needed, specify it as part of the validator's config to only apply to the specific entity and aspect. We would only expect to see this if we support multiple entities/aspects in a validator.